### ISSUE TITLE
no-unused-modules: in some cases, imports from files with capital letters in their names are not checked

### ISSUE THREAD

**Issue by danyadev** [benmosher/eslint-plugin-import#1910]
OS: Windows 10 Pro

The full eslint config and dependency versions can be viewed in the [application repository](https://github.com/danyadev/vk-desktop/tree/e63128b878fa186916eca2edf14c299b58b8f0f7). This bug is also reproduced there.

Perhaps this may also be due to the fact that the module itself is used in the `.vue` file.

Error:
```
C:\Users\danyadev\Desktop\vk-desktop\src\components\messages\chat\attachments\photosLayout.js
  97:1  error  exported declaration 'calculatePhotosLayout' not used within other modules  import/no-unused-modules
```

`calculatePhotosLayout` function is used in the same folder in `PhotosLayout.vue` file.

**Comment by ljharb** [comment]
I don't think this plugin could possibly work with `.vue` files - how would eslint-plugin-import know how to parse "not javascript"?

**Comment by danyadev** [comment]
this plugin works fine with all other `.vue` files in the project

**Comment by ljharb** [comment]
I'm not familiar with vue at all; do they use normal JS syntax for imports/exports? When you say "works fine", do you mean, it doesn't error on them, or do you mean, it reports errors inside them?

**Comment by ljharb** [comment]
To clarify the PR title - if you rename the file from `PhotosLayout.vue` to `photosLayout.vue` and update the corresponding import paths, does it work as expected?

**Comment by danyadev** [comment]
Sorry for the late answer, I left development for a while.

When changing the name of the .vue file, the problem is also reproduced:

![image](https://user-images.githubusercontent.com/35631027/104732704-aad3b980-574e-11eb-80c8-359e0601bc0e.png)

**Comment by danyadev** [comment]
.vue files are handled entirely by vue-loader and its repository [contains](https://github.com/vuejs/vue-loader/tree/next#how-it-works) a description of how it works

"works fine" means that the plugin is actually monitoring the use of exports:

![image](https://user-images.githubusercontent.com/35631027/104746954-fe4f0300-5760-11eb-90bf-9aadad61ad7c.png)
![image](https://user-images.githubusercontent.com/35631027/104747132-31919200-5761-11eb-8a9d-a0c914d34a82.png)

**Comment by TheJaredWilcurt** [comment]
Same-ish problem. Inside of a `.vue` file the plugin will do things like enforce the order of imports, but if a `.js` file exports something that is only imported in `.vue` files, it will ignore them and think that there are no imports of it.

I tried renaming the `.vue` file to be all lowercase, but that didn't change anything.

```js
// example.js
export function example () {
  return 'text';
}
```
```html
<!-- myexample.vue -->
<template>
  <div>Text</div>
</template>

<script>
import { dummy } from './example.js';

console.log(dummy());

export default {
  name: 'MyExample'
};
</script>
```

```
C:\project\example.js
  2:1  error  exported declaration 'example' not used within other modules  import/no-unused-modules
```

Again, [all other rules I'm using](https://github.com/tjw-lint/eslint-config-tjw-import/blob/main/index.js) apply to `.vue` files, it's just this one that doesn't work.

**Example project for reproduction:**

* https://github.com/TheJaredWilcurt/vite-vue3-pinia-options-api

**Comment by fatyrify** [comment]
same issue here, so this rule `import/no-unused-modules` still not working in .vue <script> files ?
